pydantic/_internal/_decorators.py

```diff
@@ -536,23 +536,7 @@ def wrapper3(value: Any, info: FieldValidationInfo) -> Any:
     )
 
 
-@overload
-def make_generic_v2_field_validator(
-    validator: FieldWrapValidatorFunction, mode: Literal['wrap']
-) -> FieldWrapValidatorFunction:
-    ...
-
-
-@overload
-def make_generic_v2_field_validator(
-    validator: OnlyValueValidator | FieldValidatorFunction, mode: Literal['before', 'after', 'plain']
-) -> FieldValidatorFunction:
-    ...
-
-
-def make_generic_v2_field_validator(
-    validator: OnlyValueValidator | FieldValidatorFunction | FieldWrapValidatorFunction, mode: str
-) -> FieldValidatorFunction | FieldWrapValidatorFunction:
+def make_generic_v2_field_or_general_validator(validator: Any, mode: str) -> Any:
```

Comment:
I gave up on the type annotations here...

Reply:
fair, fine with me. This is the good side of python types.

Reply:
side note: I do think we shouldn't use `v2` in names, that should be the default.